File: singly_linked_list/recipe-01/python/test_list_sort.py
import random

from singly_linked_list import SinglyLinkedList

N = 10

def test():
    numbers = SinglyLinkedList()
    for i in range(0, N-1):
        numbers.Append(random.randint(0,999))

    for number in numbers:
        print(number, end=" ")
    print()

    numbers.Sort();

    for number in numbers:
        print(number, end=" ")
    print()

if __name__ == "__main__":
    test()

File: singly_linked_list/recipe-01/c/test_list_sort.c
#include "integer_list.h"
#include <stdio.h>
#include <stdlib.h>

#define N 10

void print(ItemType data) 
{
    printf("%d ", data);
}

void test()
{
    SinglyLinkedList* numbers = NewList();
    for (int i = 0; i < N-1; i++) {
        ListAppend(numbers, rand() % 1000);
    }

    ListTravel(numbers, &print);
    printf("\n");

    ListSort(numbers);

    ListTravel(numbers, &print);
    printf("\n");

    FreeList(numbers);
}

int main()
{
    Init();

    test();

    return 0;
}

